dashboards/google-kubernetes-engine/gke-workloads-at-risk.json

```diff
@@ -0,0 +1,365 @@
+            }
+          },
+          "title": "MEM not set"
+        },
+        "width": 2,
+        "xPos": 0,
+        "yPos": 0
+      },
+      {
+        "height": 2,
+        "widget": {
+          "scorecard": {
+            "thresholds": [
+              {
+                "color": "RED",
+                "direction": "ABOVE",
+                "label": "",
+                "value": 0
+              }
+            ],
+            "timeSeriesQuery": {
+              "timeSeriesQueryLanguage": "fetch k8s_container\n| metric 'kubernetes.io/container/cpu/request_cores'\n| filter resource.namespace_name !~ '(kube|istio|gatekeeper|gke|gmp)-system'\n| group_by 5m, [value_request_bytes_mean: mean(value.request_cores)]\n| every 5m\n| group_by\n    [resource.cluster_name, resource.location, resource.project_id,\n     resource.namespace_name, metadata.system.top_level_controller_name]\n| { all_workloads:\n      ident\n      | group_by [], [row_count()]\n  ; best_effort_workloads:\n      filter val() == 0\n      | group_by [], [row_count()] }\n| join\n| window 5m\n| value\n    [v_0:\n       scale(div(best_effort_workloads.row_count, all_workloads.row_count), '%')]"
+            }
+          },
+          "title": "CPU not set"
+        },
+        "width": 2,
+        "xPos": 2,
+        "yPos": 0
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "**Reliability Risk**: For **memory**: kubelet can stop your Pods anytime and mark them as failed. For **CPU**: can be throttled to 0, causing your workloads to become unresponsive. [Set appropriate resource requests and limits](https://cloud.google.com/architecture/best-practices-for-running-cost-effective-kubernetes-applications-on-gke#set_appropriate_resource_requests_and_limits)",
+            "format": "MARKDOWN"
+          },
+          "title": "Best Effort workloads at risk"
+        },
+        "width": 12,
+        "xPos": 0,
+        "yPos": 2
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "**Red** - Reliability risk.  \n**Yellow** - Performance risk.",
+            "format": "MARKDOWN"
+          },
+          "title": "Dashboard Legend"
+        },
+        "width": 2,
+        "xPos": 10,
+        "yPos": 0
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "**Performance Reliability Risk**: For **memory**: when workloads are running above requests, it has the same problem as best effort pods (can be killed at any time). For **CPU**: when workloads are running above requests, slowing down the workload\n [Set appropriate resource requests and limits](https://cloud.google.com/architecture/best-practices-for-running-cost-effective-kubernetes-applications-on-gke#set_appropriate_resource_requests_and_limits)",
```

Comment:
How about

Pods using more **memory** than requested constitute a reliability risk. Like best effort pods above, they can be stopped by the kubelet anytime. When workloads consume more **CPU** than requested, they can be throttled, slowing down the workload. [Set appropriate resource requests and limits](...)

Reply:
updated